### ISSUE TITLE
Query select fires twice returning the previous value in the second time

### ISSUE THREAD

**Issue by amr-alamir** [datorama/akita#372]
## I'm submitting a...
<pre><code>
[*] Support request
[*] Bug report ??
</code></pre>

## Current behavior
In [this stackblitz](https://stackblitz.com/edit/angular-ngqgr3) I have a simple **user** `entityStore` with `activeState` and a `filter` property:
```typescript
export interface UserState extends EntityState<User>, ActiveState { 
    filter: string;
}

let initialState: UserState = {
    active: null,
    filter: '',
}
```
in `AppComponent` I get `users` from api and display it. And I filter user's `username` property via a `FormControl`

in `Component2` I display a json of the selected user

#### the problem here is:

In `Component2`, when  I select activeId I got notified 2 times, first with the correct value, then by the previous value (you can see the log in the console)
```typescript
this.userQuery.selectActiveId().subscribe( id => {
    console.log(id) // logged twice and changed to previous value
    this.user = this.userQuery.getEntity(id);
})
```
I couldn't determine the cause so if you please help me figure it out.

## Expected behavior
select property fire only once

## Minimal reproduction of the problem with instructions
[https://stackblitz.com/edit/angular-ngqgr3](https://stackblitz.com/edit/angular-ngqgr3)

**Comment by NetanelBasal** [comment]
```ts
// AppComponent
this.userService.get().subscribe(); 

this.userQuery.selectUsers$.subscribe(users => { 
  const active = users.length ? users[0].id : null;
  this.userService.updateActiveId(active); 
});


// Comp B
this.userQuery.selectActiveId().subscribe( id => {
  console.log(id); 
}) 
```
Ok, so here's what's happening:

App starts:
- `selectUsers$` emits synchronously, and updates `active` to `null`.
- `selectActiveId` emits synchronously with `null`.

So far, so good.

`userService.get()` returns:
- It calls `store.set()` which changes `active` to `null`, and updates the `users`.
- `selectUsers$` emits synchronously with the `users` and update `active` to the first user id.
- `selectActiveId` emits synchronously and recivies the new active id.

But now we have a problem. The `selectActiveId` subscriber still needs to get the emission fired as a result of the store update, which changed the `active` to `null`. That's why you see: `null`, `id`, `null`

**Comment by NetanelBasal** [comment]
You have two options:

- Add a `delay(0)` to `selectUsers$`
- Move the `active` update to the service:

```ts
    return this.http
      .get<User[]>("https://jsonplaceholder.typicode.com/users")
      .pipe(
        tap(users => {
          applyTransaction(() => {
            this.userStore.set(users);
            let found: boolean = !!users.find(
              o => o.id == this.userQuery.getActiveId()
            ); 
            let id: number = users.length
              ? found
                ? this.userQuery.getActiveId()
                : users[0].id
              : null;
            this.updateActiveId(id);
          });
        })
      );
```

**Comment by NetanelBasal** [comment]
I have added in the latest version the option to pass the active id to `set`:

```ts
this.userStore.set(users, { activeId: users[0].id });
```

**Comment by amr-alamir** [comment]
thanks. This issue was hanging for a while. It recurred several time in my project, and it drove me crazy for not being able to determine the cause.  I was just about to switch temporarily to ngrx till this issue get fixed.

The **`delay(0)`**  solution is good. It defers the execution from synchronous to asynchronous and problem solved.

I think I'll stick to `delay(0)`  solution as it fix another issue with
```ts
userStore.remove(id);
```

like you did in this fix:

> I have added in the latest version the option to pass the active id to `set`:
> 
> ```ts
> this.userStore.set(users, { activeId: users[0].id });
> ```

I think you should add similar fix to `store.remove(id)` to update the next active id

```ts
  this.userStore.remove(id, { activeId: nextId });
```

**Comment by NetanelBasal** [comment]
Let me give you a tip. It's not nice to say to an author of open-source library things like " I was just about to switch temporarily back to ngrx till this issue get fixed."

You're welcome to submit a PR and add the same functionality to the remove method.

**Comment by amr-alamir** [comment]
I don't think there a competition between state management libraries here.
But let me tell you, if you take it this way, and if it calms you down: with continuous support, Akita will outrun other libraries for sure, and it should be the **official** `@angular/state` library.

**Comment by googleHireMe** [comment]
@NetanelBasal just had the same issue and also sit with it for about 14 hours.
Could you please tell me if  `delay(0)` is good solution. Because it looks like using setTimeout(...) to fix change detection in Angular which is actually bad practice and indicates problems in my application's architecture. Does using `delay(0)` means I made architectural mistake? 
Also if the answer if 'no' I would be happy to somehow fix this bug in Akita, but for now I just don't understand the root cause, could you please clarify it to me?

**Comment by NetanelBasal** [comment]
It doesn't relate to Akita. I described in the first comment what's happened. This is how Angular and Rx work. You can choose one of the options I gave to solve it.

**Comment by googleHireMe** [comment]
@NetanelBasal  thank you for lightning fast answer!

**Comment by ketijakrista** [comment]
Ran into a similar problem with selectActive(), Subscription fired twice. The only thing that worked for me was this: 
`selectActive().pipe(skip(1))`
